### PR TITLE
Issue 2705

### LINKED ISSUE #2705: Expose RPC for creating test case categories
**Is your feature request related to a problem? Please describe.**
Creating categories  in the test case is a manual effort. It gets tedious to add when we have multiple catagories.

**Describe the solution you'd like**
There is xml rpc (tcms.rpc.api.category.filter) https://kiwitcms.readthedocs.io/en/latest/modules/tcms.rpc.api.category.html#tcms.rpc.api.category.filter. 
We can expose more functional RPC's that could Create,Update, Delete etc.

**Describe alternatives you've considered**
Manual Creation.

### PR DESCRIPTION
**Pull request by Venefyxatu**
I needed creation of Classification through the XMLRPC API and would eventually need Category creation as well. Implemented both.

Fixes #2705

### CODE REVIEW

File: tcms/rpc/api/category.py [hunk 1]
```diff
@@ -24,3 +27,27 @@ def filter(query):  # pylint: disable=redefined-builtin
...
+        :raises ValueError: if input values don't validate
+        :raises PermissionDenied: if missing *testcases.add_category* permission
+    """
+    form = CategoryForm(values)
+
+    if form.is_valid():
+        category = form.save()
+        return model_to_dict(category)
+
+    raise ValueError(form_errors_to_list(form))
```
Comment (atodorov) [inline tcms/rpc/api/category.py:53]: Looks good but needs a test. See `TestVersionCreatePermissions`.

File: tcms/rpc/api/classification.py [hunk 2]
```diff
@@ -22,3 +25,27 @@ def filter(query):  # pylint: disable=redefined-builtin
...
+        :raises ValueError: if input values don't validate
+        :raises PermissionDenied: if missing *management.add_product* permission
+    """
+    form = ClassificationForm(values)
+
+    if form.is_valid():
+        classification = form.save()
+        return model_to_dict(classification)
+
+    raise ValueError(form_errors_to_list(form))
```
Comment (atodorov) [inline tcms/rpc/api/classification.py:51]: Looks good, but needs a test. See `TestPlanTypeCreate` for example.

**Review by atodorov — CHANGES_REQUESTED** [review]
Overall looks good but need some new tests added. See the references for examples.

**Comment by Venefyxatu** [comment]
Thanks for the pointers to the tests to use for inspiration! If more is required I'll be happy to do so.

File: tcms/rpc/tests/test_classification.py [hunk 3]
```diff
@@ -45,3 +47,26 @@ def verify_api_with_permission(self):
...
+class TestClassificationCreate(APIPermissionsTestCase):
+    permission_label = "management.add_classification"
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+        self.product = ProductFactory()
+
+    def verify_api_with_permission(self):
+        result = self.rpc_client.Classification.create({"name": "API-CATEGORY",
+                                                        "product": self.product.id})
```
Comment (atodorov) [inline tcms/rpc/tests/test_classification.py:61]: Classification objects don't have a field `product`, only `name`. Looks like a copy-paste error. 

Interestingly enough the test didn't fail ?!?.


While at it adjust the name to something different, not "CATEGORY".

File: tcms/rpc/tests/test_classification.py [hunk 4]
```diff
@@ -45,3 +47,26 @@ def verify_api_with_permission(self):
...
         with self.assertRaisesRegex(ProtocolError, "403 Forbidden"):
             self.rpc_client.Classification.filter({})
+
+
+class TestClassificationCreate(APIPermissionsTestCase):
+    permission_label = "management.add_classification"
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+        self.product = ProductFactory()
```
Comment (atodorov) [inline tcms/rpc/tests/test_classification.py:57]: Once the code below is adjusted this whole method becomes unnecessary.

**Review by atodorov — CHANGES_REQUESTED** [review]
Minor changes requested. Also see isort and black linters.

File: tcms/rpc/tests/test_classification.py (hunk 3 above)
Comment (Venefyxatu) [reply tcms/rpc/tests/test_classification.py:61]: That really was too much copy/paste, yes. 
It looks like any unknown keys are just ignored - confirmed through ipython:
```
In [26]: kiwi.exec.Classification.create({'name': 'extra', 'product': 1, 'drinks': False})
Out[26]: {'id': 5, 'name': 'extra'}

In [27]: kiwi.exec.Classification.filter({'name': 'extra'})
Out[27]: [{'id': 5, 'name': 'extra'}]
```

File: tcms/rpc/api/classification.py [hunk 5]
```diff
@@ -22,3 +25,27 @@ def filter(query):  # pylint: disable=redefined-builtin
...
+    .. function:: RPC Classification.create(values)
+
+        Create a new Classification object and store it in the database.
+
+        :param values: Field values for :class:`tcms.management.models.Classification`
+        :type values: dict
+        :return: Serialized :class:`tcms.management.models.Classification` object
+        :rtype: dict
+        :raises ValueError: if input values don't validate
+        :raises PermissionDenied: if missing *management.add_product* permission
```
Comment (atodorov) [inline tcms/rpc/api/classification.py:43]: Small typo: `management.add_classification` permission!

**Review by atodorov — APPROVED** [review]

**Pull request merged**